libs/utils/reports.py

```diff
@@ -23,52 +23,49 @@ class ReportsService:
 
             Returns:
                 result (dict): {
                     'hours': [str],
-                    'detected_objects': [float],
-                    'violating_objects': [float]
+                    'detected_objects': [int],
+                    'violating_objects': [int]
                 }
-                Ex: {'dates': [00, 01, ...], 'detected_objects': [7273.0, 0.5, ...], 'violating_objects': [4920.3, 0.4, ...]}
+                Ex: {'hours': [00, 01, ...], 'detected_objects': [7273, 1, ...], 'violating_objects': [4920, 0, ...],
+                     'DetectedFaces': [1001, 0, ...], 'UsingFacemask': [406, 0, ...]
+                    }
         """
-        log_dir = os.getenv('LogDirectory')
-        dir_path = os.path.join(log_dir, camera_id, "objects_log")
-        date_range = pd.date_range(start=from_date, end=to_date)
+        log_dir = os.getenv('SourceLogDirectory')
+        dir_path = os.path.join(log_dir, camera_id, "reports")
         hours = list(range(0, 24))
         detected_objects = np.zeros(24)
         violating_objects = np.zeros(24)
         detected_faces = np.zeros(24)
         faces_with_mask = np.zeros(24)
 
-        if os.path.exists(os.path.join(dir_path, 'report.csv')):
-            iters = 0
-            for date in date_range:
-                file_path = os.path.join(dir_path, f"report_{date.strftime('%Y-%m-%d')}.csv")
-                if os.path.exists(file_path):
-                    iters += 1
-                    df = pd.read_csv(file_path).drop(['Number'], axis=1)
-                    detected_objects = detected_objects + df['DetectedObjects'].to_numpy()
-                    violating_objects = violating_objects + df['ViolatingObjects'].to_numpy()
-                    detected_faces = detected_faces + df.get('DetectedFaces', 0).to_numpy()
-                    faces_with_mask = faces_with_mask + df.get('UsingFacemask', 0).to_numpy()
-
-            if iters != 0:
-                detected_objects = detected_objects/iters
-                violating_objects = violating_objects/iters
+        if os.path.join(dir_path, f"report_{date}.csv"):
+            file_path = os.path.join(dir_path, f"report_{date}.csv")
+            if os.path.exists(file_path):
+                df = pd.read_csv(file_path).drop(['Number'], axis=1)
+                detected_objects = df['DetectedObjects'].to_numpy()
+                violating_objects = df['ViolatingObjects'].to_numpy()
+                detected_faces = df.get('DetectedFaces', 0).to_numpy()
+                faces_with_mask = df.get('UsingFacemask', 0).to_numpy()
 
         report = {
             'hours': hours,
-            'detected_objects': np.around(detected_objects).tolist(),
-            'violating_objects': np.around(violating_objects).tolist(),
-            'detected_faces': np.around(detected_faces).tolist(),
-            'faces_with_mask': np.around(faces_with_mask).tolist(),
+            'detected_objects': np.pad(
```

Comment:
Why could detected_objects have less than 24 values? Only because the day hasn't finished? Could there be missing values in the middle?

Reply:
When you request the current day data.